sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -804,7 +804,14 @@ static void _find_req_memo(gcoap_request_memo_t **memo_ptr, coap_pkt_t *src_pdu,
         } else if (coap_get_token_len(memo_pdu) == cmplen) {
             memo_pdu->token = coap_hdr_data_ptr(memo_pdu->hdr);
             if ((memcmp(src_pdu->token, memo_pdu->token, cmplen) == 0)
-                    && sock_udp_ep_equal(&memo->remote_ep, remote)) {
+                    && (
+                     sock_udp_ep_equal(&memo->remote_ep, remote) ||
+                     /* Multicast addresses are not considered in matching responses */
+                     (
+                      memo->remote_ep.family == AF_INET6 &&
+                      ipv6_addr_is_multicast((const ipv6_addr_t *)&memo->remote_ep.addr.ipv6)
```

Comment:
Want to move that to a helper function? `sock_udp_ep_is_multicast()`

Reply:
Yes, that'd be a good idea; will update when I get back to it.